posts/2020/12/jupyterlab-winter-theme.md

```diff
@@ -0,0 +1,347 @@
+improvements to the extension system. Theming is a way to extend JupyterLab and 
+benefits from those improvements.
+
+While theming is often disregarded as purely cosmetic endeavour it can greatly
+improve software. Theming can be great help for accessibility, and even the
+Jupyter team pays attention to making the default appearance accessibility aware by using
+sufficient contrast.  For users with a high visual acuity you may also choose 
+to increase the information density.
+
+Theming can also be a great way to improve communication by increasing or
+decreasing emphasis of the user interface, which can be of use for teaching, or
+presentation. Theming may also help with security, for example, by having a clear
+distinction between staging and production.
+
+Finally Theming can be a great way to express oneself, for example, by using
+a branded version of software that fits well into a context, or expressing one's artistic
+preferences or opinions. 
+
+In the following blog post, we will show you step-by-step how you can
+develop a custom theme for JupyterLab, distribute it, and take the example of the
+jupyterlab-theme-winter theme we release today to celebrate the end of 2020.
+
+<!-- TEASER_END -->
+
+# JupyterLab and Themes
+
+JupyterLab customisation can be done via what are called `Extensions`.  All
+behaviors and user interface elements of JupyterLab can be changed by providing
+extensions; this is true for elements that are added to JupyterLab, but also for
+the core components. A Theme is one of those extensions. The default light and dark
+themes are always good examples to look at if you want to understand how to build
+a theme.
+
+Generally all information about how extensions work in JupyterLab is applicable
+to a theme, though there are a number of optional steps and behaviors that are not
+necessary for themes, and a few configurations that are needed for themes.
+A lot of boilerplate can also be extracted, which makes creating most themes 
+simpler that full-fledged extensions.
+
+Let's first see what we are trying to accomplish in a screenshot of the "winter 2020" theme.
+You will have the option to choose this theme in the dropdown menu of JupyterLab:
+
+[ ![Screenshot of JupyterLab Winter Theme](/images/jupyterlab-theme-winter.png) ](/images/jupyterlab-theme-winter.png)
+
+To do so we'll need to:
+
+- Install JupyterLab
+- Create a new theme extension
+- Install this extension
+- Switch to our new theme
+- Progressively change the appearance of our JupyterLab components until we get
+  the desired outcome
+
+Optionally, once you are happy with the result, or would like to get feedback from the
+community, you can publish the theme so that people can ~~complain~~ suggest improvements
+and contribute. 
+
+# Installing JupyterLab
+
+To get started you do not need a development version of JupyterLab.  This can
+be achieved with:
+
+```bash
+$ pip install --pre \
+  jupyterlab==3.0rc14 \
+  jupyter_packaging \
+  cookiecutter
```

Comment:
This looks a bit weird, why not put it on a single line?

Reply:
single-lined.